sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java

```diff
@@ -69,4 +70,47 @@ public void start_StartedThenComplete() throws InterruptedException {
         Assert.assertTrue(listener.finishCalled);
         Assert.assertFalse(listener.errorCalled);
     }
+
+    @Test
+    public void testRequestFilters() throws Exception {
+
+        FilterCallCounter filterCallCounter = new FilterCallCounter();
+        TestReplicationListener listener = new TestReplicationListener();
+        PullReplication pullReplication = createPullReplication();
+        pullReplication.requestFilters.add(filterCallCounter);
+        Replicator replicator = ReplicatorFactory.oneway(pullReplication);
+        replicator.getEventBus().register(listener);
+        replicator.start();
+
+        while(replicator.getState() != Replicator.State.COMPLETE && replicator.getState() != Replicator.State.ERROR) {
+            Thread.sleep(50);
+        }
+
+        Assert.assertEquals(Replicator.State.COMPLETE, replicator.getState());
+        Assert.assertFalse(listener.errorCalled);
+        Assert.assertTrue(listener.finishCalled);
```

Comment:
Can `runReplicationUntilComplete` be extended to a utility method used by PushReplicatorTest too? The method can be push/pull agnostic.